tasks/elasticsearch-ssl.yml

```diff
@@ -0,0 +1,23 @@
+---
+- name: ensure certificate directory exists
+  file:
+    dest: "{{ es_ssl_certificate_path }}"
+    state: directory
+
+- name: Upload HTTP SSL/TLS certificates
+  copy:
+    src: "{{ item }}"
+    dest: "{{ es_ssl_certificate_path }}/{{ item | basename }}"
+  with_items:
+    - "{{ es_ssl_key }}"
+    - "{{ es_ssl_certificate }}"
+  when: es_enable_http_ssl|bool or es_enable_transport_ssl|bool
+
+- local_action: stat path="{{ role_path }}/files/{{ es_ssl_certificate_authority }}"
+  register: es_cafile
+
+- name: Upload SSL Certificate Authority
+  copy:
+    src: "{{ es_ssl_certificate_authority }}"
+    dest: "{{ es_ssl_certificate_path }}/{{ es_ssl_certificate_authority | basename }}"
+  when: es_cafile.stat.exists|bool and es_cafile.stat.isreg|bool
```

Comment:
> Is there any specific reason we have these checks on `es_ssl_certificate_authority` file but not on `es_ssl_key`, and `es_ssl_certificate`?

I guess it's optional, but also strange it would be checking in the `role_path` and not the `playbook_dir` where the user defined keys and certs are more likely to be.

> If we need these checks we could add them to [`elasticsearch-parameters.yml`](https://github.com/elastic/ansible-elasticsearch/blob/master/tasks/elasticsearch-parameters.yml) so Ansible would fail at the beginning if files are missing.

It would be difficult to check for these in [`elasticsearch-parameters.yml`](https://github.com/elastic/ansible-elasticsearch/blob/master/tasks/elasticsearch-parameters.yml) as the paths could be full or relative and the `stat` module only checks full paths. Probably better removing the file check and updating the conditional on the copy to check that es_ssl_certificate_authority is not empty/defined.

Reply:
There wasn't any specific reason that I only checked certain variables / paths, instead of all of them. This emerged from the external tooling I was using: either all files were present or none were present, so checking for a single file was sufficient.

There's no harm in checking all of them, to be sure.